core/iwasm/compilation/aot_emit_gc.c

```diff
@@ -104,4 +136,1476 @@ aot_call_aot_create_func_obj(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
+
+    BUILD_ISNULL(struct_obj, cmp, "cmp_struct_obj");
+    if (!aot_emit_exception(comp_ctx, func_ctx, EXCE_NULL_GC_REF, true, cmp,
+                            check_struct_obj_succ))
+        goto fail;
+
+    if (!aot_call_wasm_struct_obj_set_field(comp_ctx, func_ctx, struct_obj,
+                                            I32_CONST(field_idx), field_value))
+        goto fail;
+
+    return true;
+fail:
+    return false;
+}
+
+static bool
+aot_call_wasm_array_obj_new(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
+                            LLVMValueRef rtt_type, LLVMValueRef array_len,
+                            LLVMValueRef array_elem, LLVMValueRef *array_obj)
+{
+    LLVMValueRef param_values[4], func, value, res, array_elem_ptr;
+    LLVMTypeRef param_types[4], ret_type, func_type, func_ptr_type;
+
+    if (!(array_elem_ptr = LLVMBuildAlloca(
+              comp_ctx->builder, LLVMTypeOf(array_elem), "array_elem_ptr"))) {
+        aot_set_last_error("llvm build alloca failed.");
+        goto fail;
+    }
+    if (!LLVMBuildStore(comp_ctx->builder, array_elem, array_elem_ptr)) {
+        aot_set_last_error("llvm build store failed.");
+        goto fail;
+    }
+    if (!(array_elem_ptr = LLVMBuildBitCast(comp_ctx->builder, array_elem_ptr,
+                                            INT8_PTR_TYPE, "array_elem_ptr"))) {
+        aot_set_last_error("llvm build bitcast failed.");
+        goto fail;
+    }
+
+    param_types[0] = INT8_PTR_TYPE;
+    param_types[1] = INT8_PTR_TYPE;
+    param_types[2] = I32_TYPE;
+    param_types[3] = INT8_PTR_TYPE;
+    ret_type = GC_REF_TYPE;
+
+    GET_AOT_FUNCTION(wasm_array_obj_new, 4);
+
+    /* Call function wasm_array_obj_new() */
+    param_values[0] = func_ctx->exec_env;
+    param_values[1] = rtt_type;
+    param_values[2] = array_len;
+    param_values[3] = array_elem_ptr;
+    if (!(res = LLVMBuildCall2(comp_ctx->builder, func_type, func, param_values,
+                               4, "call"))) {
+        aot_set_last_error("llvm build call failed.");
+        goto fail;
+    }
+
+    *array_obj = res;
+    return true;
+fail:
+    return false;
+}
+
+static bool
+aot_call_wasm_array_set_elem(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
+                             LLVMValueRef array_obj, LLVMValueRef elem_idx,
+                             LLVMValueRef array_elem)
+{
+    LLVMValueRef param_values[3], func, value, array_elem_ptr;
+    LLVMTypeRef param_types[3], ret_type, func_type, func_ptr_type;
+
+    if (!(array_elem_ptr = LLVMBuildAlloca(
+              comp_ctx->builder, LLVMTypeOf(array_elem), "array_elem_ptr"))) {
+        aot_set_last_error("llvm build alloca failed.");
+        goto fail;
+    }
+    if (!LLVMBuildStore(comp_ctx->builder, array_elem, array_elem_ptr)) {
+        aot_set_last_error("llvm build store failed.");
+        goto fail;
+    }
+    if (!(array_elem_ptr = LLVMBuildBitCast(comp_ctx->builder, array_elem_ptr,
```

Comment:
Same as above, directly use load/store IRs to access array element to improve performance.